Kernel/Problem.hpp

```diff
@@ -67,23 +70,70 @@ class Problem {
   bool hadIncompleteTransformation() const { return _hadIncompleteTransformation; }
   void reportIncompleteTransformation() { _hadIncompleteTransformation = true; }
 
-  typedef DHMap<unsigned,bool> TrivialPredicateMap;
-  void addTrivialPredicate(unsigned pred, bool assignment);
+  enum class IntereferenceKind : unsigned char {
+    FUN_DEF = 0,
+    PRED_DEF = 1,
+    GLOB_FLIP = 2,  // backs up random_polarities
+    COND_FLIP = 3,  // backs up partially  predicates (Plaisted-Greenbaum-style p<=>q reduced to, e.g. p=>q) and eliminated blocked clauses
+  };
+
+  /* Intereference (satisfiability-preserving-only transformations) recording */
+  struct Intereference {
```

Comment:
Typo - "intereference"/"interference" - normally I'd let it slide in the interest of authenticity, but I think this would drive me crazy, sorry.